tests/jeos/firstrun.pm

```diff
@@ -58,6 +58,25 @@ sub verify_mounts {
     assert_script_run('mount -fva');
 }
 
+sub verify_hypervisor {
+    my $virt = script_output('systemd-detect-virt');
+
+    return 0 if (
+        is_qemu && $virt =~ /(qemu|kvm)/ ||
+        is_s390x && $virt =~ /zvm/ ||
+        is_hyperv && $virt =~ /microsoft/ ||
+        is_vmware && $virt =~ /vmware/);
+
+    die("Unknown hypervisor: $virt");
+}
+
+sub verify_norepos {
+    my $ret = script_run "zypper -n refresh";
+
+    # Check ZYPPER_EXIT_NO_REPOS
+    die("Image should not contain any repos after first boot") if ($ret != 6);
```

Comment:
This check is only valid for unregistered SLES JeOS images but not for openSUSE images.